### DISCUSSION TITLE
test the icub-gazebo-grasping locally than the sandbox

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/icub-gazebo-grasping-sandbox#52] [Q&A] [answered]
Hi, 

i tried to install the app on my ubuntu system rather than test it in the sandbox, but there is some make problem.

Thanks.

```
icub@icubsrv:~/workspace/icub-gazebo-grasping-sandbox/build$ make
[ 33%] Built target icub-gazebo-grasping-sandbox-model-mover
[ 50%] Building CXX object src/CMakeFiles/icub-gazebo-grasping-sandbox.dir/main.cpp.o
/home/icub/workspace/icub-gazebo-grasping-sandbox/src/main.cpp:20:10: fatal error: yarp/os/Network.h: No such file or directory
   20 | #include <yarp/os/Network.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/icub-gazebo-grasping-sandbox.dir/build.make:71: src/CMakeFiles/icub-gazebo-grasping-sandbox.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:193: src/CMakeFiles/icub-gazebo-grasping-sandbox.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

**Comment by pattacini** [comment]
Hi @jimingda 

This very depends on how your system is configured for building.
Probably, it's not in line for compiling against YARP, which should be found according to this line: https://github.com/robotology/icub-gazebo-grasping-sandbox/blob/master/src/CMakeLists.txt#L8.

Sorry, but this sandbox is meant to run as it is. I'm not providing assistance for deploying it on a different system.

If you aim to run it locally, you can refer to these instructions, which rely on the use of docker:
- https://github.com/robotology/icub-gazebo-grasping-sandbox/blob/master/dockerfiles/README.md

**Reply by jimingda** [reply]
Thanks @pattacini 

Follow the instruction, i run the app in the docker and got some problem when `cmake -S . -B build`
```
gitpod@59ada756cc1f:/workspace/icub-gazebo-grasping-sandbox$ sudo cmake -S . -B build
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:11 (find_package):
  By not providing "FindYCM.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "YCM", but
  CMake did not find one.

  Could not find a package configuration file provided by "YCM" with any of
  the following names:

    YCMConfig.cmake
    ycm-config.cmake

  Add the installation prefix of "YCM" to CMAKE_PREFIX_PATH or set "YCM_DIR"
  to a directory containing one of the above files.  If "YCM" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
See also "/workspace/icub-gazebo-grasping-sandbox/build/CMakeFiles/CMakeOutput.log".
```

**Reply by pattacini** [reply, marked as answer]
Using `sudo` wasn't required in the first place and can explain why the build fails.
I've just made a test cd'ing `home` first and it worked.

The correct instructions are thus:
```console
cd ~
git clone https://github.com/robotology/icub-gazebo-grasping-sandbox.git
cd icub-gazebo-grasping-sandbox
cmake -S . -B build
cmake --build build/ --target install
```

At any rate, I think I can make user gitpod the owner of `/workspace`.

**Reply by pattacini** [reply]
Fix proposed via:
- #53

I'm regenerating the image. See https://github.com/robotology/icub-gazebo-grasping-sandbox/actions/runs/6548697949

**Reply by pattacini** [reply]
Release [v2.10.4](https://github.com/robotology/icub-gazebo-grasping-sandbox/releases/tag/v2.10.4) solved the problem via the above PR.

**Reply by jimingda** [reply]
Thanks @pattacini 

I succeed to install the app on ubuntu and run it from `yarpmanager`. 
```
icub@icubsrv:~/workspace/icub-gazebo-grasping-sandbox$ yarp rpc /icub-grasp/rpc
>>go on
Response: [ok]
>>
```

![2023-10-24_17-16](https://github.com/robotology/icub-gazebo-grasping-sandbox/assets/7019692/db72909f-4e60-4e2d-a119-4ecfaf409ede)
